src/node/timer.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/logger.h"
+#include "ds/spinlock.h"
+
+#include <chrono>
+#include <functional>
+#include <list>
+
+namespace ccf
+{
+  using TimerCallback = std::function<bool()>;
+
+  /**
+   * A timer class to trigger actions periodically.
+   *
+   * A shared_ptr to a new Timer is returned by Timers::new_timer(period, cb).
+   * The timer is only active when it is started (start()) and marked as expired
+   * when the callback is triggered. If the callback returns true, the timer
+   * continues ticking. Otherwise, the timer expires and will tick again only
```

Comment:
I'm confused by this behaviour. Why does it keep ticking after the callback has been called? If it _reset_ the timer so the callback would be called again, after another period of the initial timeout, I could understand. But it seems a post-callback `STARTED` timer is just counting time for no purpose?